src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/YesNo.java

```diff
@@ -25,7 +29,23 @@ public String getValue() {
     }
 
     public boolean toBoolean() {
-        return YES.value.equals(value);
+        return this.bool;
+    }
+
+    public static YesNo isYesOrNo(boolean isYes) {
+        return isYes ? YES : NO;
+    }
+
+    public static YesNo isYesOrNo(String yesNo) {
+        return isYesOrNo(isYes(yesNo));
+    }
+
+    public static YesNo isYesOrNo(YesNo yesNo) {
+        return isYesOrNo(isYes(yesNo));
+    }
+
+    public static YesNo isYesOrNoOrNull(String yesNo) {
+        return nonNull(yesNo) && !yesNo.trim().isEmpty() ? isYesOrNo(yesNo) : null;
     }
 
```

Comment:
Only the first isYesOrNo method is needed because theres only either 1 or 2 values possible. so we can automatically assume if its not going to be yes then it has to be no. I personally think we shouldnt be using yes or no and just use boolean yes or no but thats a different discussion for another day.